pkg/api/frontendsettings.go

```diff
@@ -112,11 +113,16 @@ func (hs *HTTPServer) getFSDataSources(c *models.ReqContext, enabledPlugins *plu
 	// the datasource table)
 	for _, ds := range hs.PluginManager.DataSources() {
 		if ds.BuiltIn {
-			dataSources[ds.Name] = map[string]interface{}{
+			info := map[string]interface{}{
 				"type": ds.Type,
 				"name": ds.Name,
 				"meta": hs.PluginManager.GetDataSource(ds.Id),
 			}
+			if ds.Name == grafana.DatasourceName {
```

Comment:
@wbrowne -- I put this back so that the front end can treat this like any other data source, and will work fine as we move to looking things up by uid rather than name.  See: https://github.com/grafana/grafana/pull/33817

The more normal this data source can be, the better :)

Reply:
Yeah makes sense 👍 Do weird things start to happen on the frontend if we set uid and id here for all built in data sources?

Reply:
In #33817, we set the id/uid equal to the name when not set, so likely a good idea to just do it from the initial response.  For "dashboard" and "mixed" it should never get sent to the backend, but who knows